File: mifosng-android/src/main/java/com/mifos/mifosxdroid/offline/syncgrouppayloads/SyncGroupPayloadsMvpView.java
package com.mifos.mifosxdroid.offline.syncgrouppayloads;

import com.mifos.mifosxdroid.base.MvpView;
import com.mifos.objects.group.GroupPayload;

import java.util.List;

/**
 * Created by Rajan Maurya on 19/07/16.
 */
public interface SyncGroupPayloadsMvpView extends MvpView {

    void showOfflineModeDialog();

    void showGroups(List<GroupPayload> groupPayloads);

    void showGroupSyncResponse();

    void showGroupSyncFailed(String errorMessage);

    void showPayloadDeletedAndUpdatePayloads(List<GroupPayload> groupPayloads);

    void showGroupPayloadUpdated(GroupPayload groupPayload);

    void showError(int stringId);
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/offline/syncgrouppayloads/SyncGroupPayloadsMvpView.kt
package com.mifos.mifosxdroid.offline.syncgrouppayloads

import com.mifos.mifosxdroid.base.MvpView
import com.mifos.objects.group.GroupPayload

/**
 * Created by Rajan Maurya on 19/07/16.
 */
interface SyncGroupPayloadsMvpView : MvpView {
    fun showOfflineModeDialog()
    fun showGroups(groupPayloads: List<GroupPayload>)
    fun showGroupSyncResponse()
    fun showGroupSyncFailed(errorMessage: String)
    fun showPayloadDeletedAndUpdatePayloads(groupPayloads: List<GroupPayload>)
    fun showGroupPayloadUpdated(groupPayload: GroupPayload)
    fun showError(stringId: Int)
}